src/install/index.js

```diff
@@ -115,29 +115,50 @@ const warmViews = (deployDoc) => {
   const writeProgress = () => {
     return DB.activeTasks()
       .then(tasks => {
-        // TODO: make the write-over better here:
-        // Order these sensibly so the UI doesn't have to
-        // If it's new add it
-        // If it was already there update it
-        // If it's gone make its progress 100%
         const relevantTasks = tasks.filter(task =>
           task.type === 'indexer' && task.design_document.includes(':staged:'));
 
         const entry = deployDoc.log[deployDoc.log.length - 1];
 
-        entry.indexers = relevantTasks;
+        entry.indexers = updateIndexers(entry.indexers, relevantTasks);
 
         return utils.update(deployDoc);
       })
       .then(() => process.stdout.write('.'));
   };
 
+  const updateIndexers = (indexers, activeTasks) => {
+    indexers = indexers || [];
+
+    indexers.forEach(indexer => Object.keys(indexer.tasks).forEach(pid => indexer.tasks[pid] = 100));
+
+    activeTasks.forEach(task => {
+      let indexer = indexers.find(indexer => indexer.design_document === task.design_document);
+      if (!indexer) {
+        indexer = {
+          design_document: task.design_document,
+          tasks: {}
+        };
+        indexers.push(indexer);
+      }
+      indexer.tasks[task.pid] = task.progress;
```

Comment:
for completeness / unlikely edge case reduction we should probably index by a composite key of the node and the pid, to avoid situations with multiple servers indexing by the same pid.